generator/dart/packages/generated/google_cloud_protobuf/lib/protobuf.dart

```diff
@@ -0,0 +1,819 @@
+/// 'type.googleapis.com/full.type.name' as the type URL and the unpack
+/// methods only use the fully qualified type name after the last '/'
+/// in the type URL, for example "foo.bar.com/x/y.z" will yield type
+/// name "y.z".
+/// 
+/// JSON
+/// ====
+/// The JSON representation of an `Any` value uses the regular
+/// representation of the deserialized, embedded message, with an
+/// additional field `@type` which contains the type URL. Example:
+/// 
+///     package google.profile;
+///     message Person {
+///       string first_name = 1;
+///       string last_name = 2;
+///     }
+/// 
+///     {
+///       "@type": "type.googleapis.com/google.profile.Person",
+///       "firstName": <string>,
+///       "lastName": <string>
+///     }
+/// 
+/// If the embedded message type is well-known and has a custom JSON
+/// representation, that representation will be embedded adding a field
+/// `value` which holds the custom JSON in addition to the `@type`
+/// field. Example (for message [google.protobuf.Duration][]):
+/// 
+///     {
+///       "@type": "type.googleapis.com/google.protobuf.Duration",
+///       "value": "1.212s"
+///     }
+/// 
+class Any {
+
+  /// A URL/resource name that uniquely identifies the type of the serialized
+  /// protocol buffer message. This string must contain at least
+  /// one "/" character. The last segment of the URL's path must represent
+  /// the fully qualified name of the type (as in
+  /// `path/google.protobuf.Duration`). The name should be in a canonical form
+  /// (e.g., leading "." is not accepted).
+  /// 
+  /// In practice, teams usually precompile into the binary all types that they
+  /// expect it to use in the context of Any. However, for URLs which use the
+  /// scheme `http`, `https`, or no scheme, one can optionally set up a type
+  /// server that maps type URLs to message definitions as follows:
+  /// 
+  /// * If no scheme is provided, `https` is assumed.
+  /// * An HTTP GET on the URL must yield a [google.protobuf.Type][]
+  ///   value in binary format, or produce an error.
+  /// * Applications are allowed to cache lookup results based on the
+  ///   URL, or have them precompiled into a binary to avoid any
+  ///   lookup. Therefore, binary compatibility needs to be preserved
+  ///   on changes to types. (Use versioned type names to manage
+  ///   breaking changes.)
+  /// 
+  /// Note: this functionality is not currently available in the official
+  /// protobuf release, and it is not used for type URLs beginning with
+  /// type.googleapis.com. As of May 2023, there are no widely used type server
+  /// implementations and no plans to implement one.
+  /// 
+  /// Schemes other than `http`, `https` (or the empty scheme) might be
+  /// used with implementation specific semantics.
+  /// 
+  final String? typeUrl;
+
+  /// Must be a valid serialized protocol buffer of the above specified type.
+  final Uint8List? value;
```

Comment:
When using JSON over HTTP we will receive a JSON object.  Are you planning to store just the JSON string here?  Or some other representation?

We will also need to implement helpers to store and extract any message into an `Any`, including the well-known types (which use custom representations over JSON).  Most likely you will want to hand-craft the `Any` class.

Reply:
> When using JSON over HTTP we will receive a JSON object. Are you planning to store just the JSON string here? Or some other representation?

TBD? This is the representation for `api.BYTES_TYPE`. Logically that would be `Uint8List` in Dart. It sounds like we may want some other custom type instead (`ProtoBuffer`?), which encodes / decodes to json, and can convert itself to Uint8List when asked.

> We will also need to implement helpers to store and extract any message into an `Any`, including the well-known types (which use custom representations over JSON). Most likely you will want to hand-craft the `Any` class.

I don't exactly know what `Any` is for, but I'll drop down to just including FieldMask in this file for now, and plan to tackle `Any` as a separate PR.